com.microsoft.mrtk.core/Utilities/ComponentCache.cs

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit
+{
+
+    /// <summary>
+    /// A static helper class which caches instances of a component of type T.
+    /// This is called in place of calling FindFirstObjectByType and FindObjectsByType to improve performance.
+    /// </summary>
+    public static class ComponentCache<T> where T : Component
+    {
+        private static T cacheFirstInstance = null;
+        private static T[] cacheList = null;
+
+        /// <summary>
+        /// Finds the first instance of an object of the given type. 
+        /// </summary>
+        /// <param name="includeInactive">Specifies whether the search should include inactive objects.</param>
+        /// <returns>
+        /// The first instance of an object of the given type. 
+        /// </returns>
+        public static T FindFirstInstance(FindObjectsInactive includeInactive = FindObjectsInactive.Exclude)
```

Comment:
Remove parameter here.  This may result in invalid cached results if one caller allows inactive components, and another only wants active components.

Maybe rename this function to `FindFirstActiveInstance`?

Reply:
good catch!!